File: Cmdlets/Helpers/LookopHelper.vb
Imports System.Collections.Generic

Module LookopHelper

	Public Function GetCountableList(Of TKey, TValue)(lookup As ILookup(Of TKey, TValue)) As IEnumerable(Of LookupCounterItem(Of TKey, TValue))
		Dim countableItems As New List(Of LookupCounterItem(Of TKey, TValue))

		For Each item In lookup
			countableItems.Add(New LookupCounterItem(Of TKey, TValue)(item.Key, item.Count, item))
		Next

		Return countableItems
	End Function

End Module

File: Common/Helpers/LookopHelper.cs
using System.Collections.Generic;
using System.Linq;
using TIKSN.Cmdlets.Common.Models;

namespace TIKSN.Cmdlets.Common.Helpers
{
    public static class LookopHelper
    {
        public static IEnumerable<LookupCounterItem<TKey, TValue>> GetCountableList<TKey, TValue>(ILookup<TKey, TValue> lookup)
        {
            List<LookupCounterItem<TKey, TValue>> countableItems = new List<LookupCounterItem<TKey, TValue>>();

            foreach (var item in lookup)
            {
                countableItems.Add(new LookupCounterItem<TKey, TValue>(item.Key, item.Count(), item));
            }

            return countableItems;
        }
    }
}